lib/datasource/git-refs/base.ts

```diff
@@ -0,0 +1,62 @@
+import simpleGit from 'simple-git';
+import { cache } from '../../util/cache/package/decorator';
+import { getRemoteUrlWithToken } from '../../util/git/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig } from '../types';
+import type { RawRefs } from './types';
+
+// TODO: extract to a separate directory structure (#10532)
+export abstract class GitDatasource extends Datasource {
```

Comment:
I don't see any benefit to extend from this class, as we only call one static method of it? 🤔

Reply:
Right now, the main benefit is the shared common cache in the same cache namespace. Yes, I could do this with a single function, but I've written this now, and it allows us to share more common code between git refs and git tags easily in the future.

Reply:
I'm ok with having a shared class, bu i think we don't need to extend from it, as all static calls will be directly to base class, so there is nothing which can be overridden by extending.

So my idea is simlply use this class in a static manner, no datasource base class.

Then the other git datasources can call the static methods as usual. They don't need to extend this class to be able to call public statuic methods. It extending makes only sense for non-static member.